packages/widget-react/README.md

```diff
@@ -182,6 +192,73 @@ function App() {
 }
 ```
 
+#### Custom Identity Provider
+
+In order to make Custom Identity Provider work, you'll have to properly implement and provide a set of following methods:
+
+- `getToken` - resolving [Chat Widget token]('https://developers.livechat.com/docs/extending-chat-widget/custom-identity-provider#chat-widget-token'). If you want to cache the token, this should return the cached token instead of a fresh request to https://accounts.livechat.com/customer/token endpoint.
+- `getFreshToken` - resolving [Chat Widget token]('https://developers.livechat.com/docs/extending-chat-widget/custom-identity-provider#chat-widget-token'). This should always make a call for a fresh token from https://accounts.livechat.com/customer/token endpoint.
+- `hasToken` - resolving boolean. It determines whether a token has been acquired.
+- `invalidate` - resolving nothing. When called, it should remove the current token. There is no need to do anything else as a new token will be requested by getFreshToken afterwards.
+
+##### Example usage
+
+```ts
+import { LiveChatWidget } from '@livechat/widget-react'
+
+const customIdentityProvider = () => {
+  const baseAPI = 'YOUR_API_URL'
+  const userId = '30317220-c72d-11ed-2137-0242ac120002'
+
+  const getToken = async () => {
+    const response = await fetch(`${baseAPI}/getToken/${userId}`)
+
+    const token = await response.json()
+    console.log('getToken', token)
+    return token
+  }
+
+  const getFreshToken = async () => {
+    const response = await fetch(`${baseAPI}/getFreshToken/${userId}`)
+
+    const token = await response.json()
+    console.log('getFreshToken, token')
+    return token
+  }
+
+  const hasToken = async () => {
+    const response = await fetch(`${baseAPI}/hasToken/${userId}`)
+    const data = await response.json()
+    return data
+  }
+
+  const invalidateToken = async () => {
+    const response = await fetch(`${baseAPI}/invalidate/${userId}`)
+    const data = await response.json()
+    console.log(data)
+  }
+
+  return {
+    getToken,
+    getFreshToken,
+    hasToken,
+    invalidate: invalidateToken,
+  }
+}
+
+function App() {
+  return (
+    <LiveChatWidget
+      license="12345678"
+      visibility="maximized"
+      customIdentityProvider={customIdentityProvider}
```

Comment:
Because the change of `customIdentityProvider` reference causes reinitialization of the whole widget the only proper way of passing CIP then without an infinite loop is to wrap it in `useCallback`. We should specify proper usage in the documentation.

I wonder those if it will not be still a confusing and easy way of falling into the trap. IMO we could just not reinitialize widget when CIP is changed. Its a provider function of customer auth module, the need for replacing it on the fly is pretty rare and that way we would be able to avoid this confusing behaviour

Reply:
This should also be taken care of in all frameworks

Reply:
adjusted in https://github.com/livechat/chat-widget-adapters/pull/72/commits/b9cc0ac889a167f52cbf95e18d08a0cca562e9d4